framework/deproxy_server.py

```diff
@@ -109,10 +112,21 @@ def handle_read(self):
             if response:
                 dbg(self, 4, "Send response:", prefix="\t")
                 tf_cfg.dbg(5, response)
-                self._cur_responses_list.append(response)
+                resp_length = (
+                    len(response)
+                    if not self._server.send_half_of_the_response
+                    else int(len(response) / 2)
+                )
+                self._cur_responses_list.append(response[:resp_length])
```

Comment:
Let's rename `_cur_responses_list` to `_data_to_send_buffer`. Now we store pieces of data, not complete responses.